app/models/chargeback.rb

```diff
@@ -74,6 +59,25 @@ def self.build_results_for_report_chargeback(options)
     [data.map { |r| new(r.last) }]
   end
 
+  def self.key_and_fields(metric_rollup_record, interval, tz)
+    ts_key = get_group_key_ts(metric_rollup_record, interval, tz)
+
+    key, extra_fields = if @options[:groupby_tag].present?
+                          get_tag_keys_and_fields(metric_rollup_record, ts_key)
+                        else
+                          get_keys_and_extra_fields(metric_rollup_record, ts_key)
+                        end
+
+    [key, date_fields(metric_rollup_record, interval, tz).merge(extra_fields)]
+  end
+
+  def self.date_fields(metric_rollup_record, interval, tz)
+    start_ts, end_ts, display_range = get_time_range(metric_rollup_record, interval, tz)
+
+    {"start_date" => start_ts, "end_date" => end_ts, "display_range" => display_range, "interval_name" => interval,
```

Comment:
Small nit, but I think that would be more readable if each k/v pair was on a separate line.